specification/purview/Azure.Analytics.Purview.DataMap/routes.tsp

```diff
@@ -0,0 +1,2084 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/http";
+import "@typespec/versioning";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.Core.Traits.Private;
+using Azure.ClientGenerator.Core;
+
+@client({
+  name: "PurviewDataMapClient",
+  service: PurviewDataMap,
+})
+namespace PurviewDataMap;
+
+
+@doc("The ApiVersion query parameter.")
+model ApiVersionParameter {
+  @query("api-version")
+  @minLength(1)
+  @doc("The API version to use for this operation.")
+  apiVersion?: string;
+}
+
+@doc("The ApiVersion query parameter for DataMap service.")
+model DataMapApiVersionParameter {
+  @query("api-version")
+  @minLength(1)
+  @doc("The API version to use for this operation.")
+  apiVersion: string;
+}
+
+model OkResponse<T> {
+  @statusCode
+  status: 200;
+  @body
+  data: T;
+}
+
+@doc("Operation for Atlas service")
+op AtlasBaseOperation<
```

Comment:
instead of defining this template through basic TypeSpec you should leverage the Azure.Core.Foundations.Operation template